plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1151,7 +1155,7 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
                     session,
                     handle.getComment(),
                     handle.getProtocolEntry());
-            appendAddFileEntries(transactionLogWriter, dataFileInfos, handle.getPartitionedBy(), columnNames, true);
+            appendAddFileEntries(transactionLogWriter, dataFileInfos, physicalPartitionNames, columnNames, true);
```

Comment:
What about a couple lines up in the `appendTableEntries` call, should we use physicalPartitionNames there too?

Reply:
Do you mean `handle.getPartitionedBy()` for `partitionColumns` in `metaData`? The current implementation is same as OSS Delta Lake. They store non-physical partition column names in the field.